File: DecompiledGameScripts/cyberpunk/devices/exitLight/exitLight.swift

public class ExitLight extends ElectricLight {

  protected cb func OnRequestComponents(ri: EntityRequestComponentsInterface) -> Bool {
    super.OnRequestComponents(ri);
  }

  protected cb func OnTakeControl(ri: EntityResolveComponentsInterface) -> Bool {
    super.OnTakeControl(ri);
    this.m_controller = EntityResolveComponentsInterface.GetComponent(ri, n"controller") as ExitLightController;
  }

  public const func GetDevicePS() -> ref<ExitLightControllerPS> {
    return this.GetController().GetPS();
  }

  private const func GetController() -> ref<ExitLightController> {
    return this.m_controller as ExitLightController;
  }

  protected cb func OnHitEvent(hit: ref<gameHitEvent>) -> Bool {
    super.OnHitEvent(hit);
  }

  protected func TurnOnDevice() -> Void {
    this.TurnGreen();
  }

  protected func TurnOffDevice() -> Void {
    this.TurnRed();
  }

  protected func ReactToHit(hit: ref<gameHitEvent>) -> Void {
    super.ReactToHit(hit);
  }

  private final func TurnRed() -> Void {
    this.SetMeshAppearance(n"red");
  }

  private final func TurnGreen() -> Void {
    this.SetMeshAppearance(n"green");
  }
}

File: Definitions/Game_Definitions/classes/ExitLight.lua
---@meta
---@diagnostic disable

---@class ExitLight : ElectricLight
ExitLight = {}

---@return ExitLight
function ExitLight.new() return end

---@param props table
---@return ExitLight
function ExitLight.new(props) return end

---@param hit gameeventsHitEvent
---@return Bool
function ExitLight:OnHitEvent(hit) return end

---@param ri entEntityRequestComponentsInterface
---@return Bool
function ExitLight:OnRequestComponents(ri) return end

---@param ri entEntityResolveComponentsInterface
---@return Bool
function ExitLight:OnTakeControl(ri) return end

---@return ExitLightController
function ExitLight:GetController() return end

---@return ExitLightControllerPS
function ExitLight:GetDevicePS() return end

---@param hit gameeventsHitEvent
function ExitLight:ReactToHit(hit) return end

function ExitLight:TurnGreen() return end

function ExitLight:TurnOffDevice() return end

function ExitLight:TurnOnDevice() return end

function ExitLight:TurnRed() return end

